op-batcher/batcher/driver.go

```diff
@@ -479,6 +521,32 @@ func (l *BatchSubmitter) safeL1Origin(ctx context.Context) (eth.BlockID, error)
 	return status.SafeL2.L1Origin, nil
 }
 
+// cancelBlockingTx creates an empty transaction of appropriate type to cancel out the incompatible
+// transaction stuck in the txpool. In the future we might send an actual batch transaction instead
+// of an empty one to avoid wasting the tx fee.
+func (l *BatchSubmitter) cancelBlockingTx(queue *txmgr.Queue[txID], receiptsCh chan txmgr.TxReceipt[txID]) {
+	txdata := txData{
+		frames: []frameData{
+			frameData{
+				data: []byte{},
+				id: frameID{
+					chID:        cancelID,
```

Comment:
I don't know offhand, is there any danger to having ID collisions if multiple channels end up getting cancelled?

Reply:
Channels aren't getting cancelled, a lingering tx of incompatible type is what we're trying to cancel (via replacement).  There can only be one cancellation tx in flight at any given moment, so there should be no collisions from using this special ID here. (If I'm understanding you correctly.)